src/main/java/com/blamejared/crafttweaker/impl/predicate/AdvancementPredicate.java

```diff
@@ -0,0 +1,71 @@
+package com.blamejared.crafttweaker.impl.predicate;
+
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import it.unimi.dsi.fastutil.objects.Object2BooleanMap;
+import it.unimi.dsi.fastutil.objects.Object2BooleanOpenHashMap;
+import net.minecraft.advancements.criterion.PlayerPredicate;
+import org.openzen.zencode.java.ZenCodeType;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+@ZenRegister
+@ZenCodeType.Name("crafttweaker.api.predicate.AdvancementPredicate")
+@Document("vanilla/api/predicate/AdvancementPredicate")
+public final class AdvancementPredicate implements IVanillaWrappingPredicate<PlayerPredicate.IAdvancementPredicate> {
+    private final Map<String, Boolean> criteria;
+
+    private TriState passed;
+
+    public AdvancementPredicate() {
+        this.criteria = new LinkedHashMap<>();
+        this.passed = TriState.UNSET;
+    }
+
+    @ZenCodeType.Method
+    public AdvancementPredicate withCompleted() {
+        this.passed = TriState.TRUE;
+        return this;
+    }
+
+    @ZenCodeType.Method
+    public AdvancementPredicate withInProgress() {
+        this.passed = TriState.FALSE;
+        return this;
+    }
+
+    @ZenCodeType.Method
+    public AdvancementPredicate withPassedCriterion(final String name) {
+        this.criteria.put(name, true);
+        return this;
+    }
+
+    @ZenCodeType.Method
+    public AdvancementPredicate withInProgressCriterion(final String name) {
+        this.criteria.put(name, false);
+        return this;
+    }
+
+    public boolean isAny() {
+        return this.passed.isUnset() && this.criteria.isEmpty();
+    }
+
+    @Override
+    public net.minecraft.advancements.criterion.PlayerPredicate.IAdvancementPredicate toVanillaPredicate() {
+        if (!this.passed.isUnset() && !this.criteria.isEmpty()) {
+            throw new IllegalStateException("An advancement can be checked only for full completion or criteria, not both");
+        }
+        if (!this.passed.isUnset()) {
+            return new net.minecraft.advancements.criterion.PlayerPredicate.CompletedAdvancementPredicate(this.passed.toBoolean());
```

Comment:
You probably don't need the qualified name, since you already imported the type, so it should take that instead of the type in the same package.
Like, your `implements` block uses the short name 😉